tools/run_tests/build_python.sh

```diff
@@ -33,44 +33,70 @@ set -ex
 # change to grpc repo root
 cd $(dirname $0)/../..
 
-TOX_PYTHON_ENV="$1"
-PY_VERSION="${TOX_PYTHON_ENV: -2}"
+PYTHON=${1:-python2.7}
+VENV=${2:-py27}
+VENV_RELATIVE_PYTHON=${3:-bin/python}
+TOOLCHAIN=${4:-unix}
 
 ROOT=`pwd`
-export LD_LIBRARY_PATH=$ROOT/libs/$CONFIG
-export DYLD_LIBRARY_PATH=$ROOT/libs/$CONFIG
-export PATH=$ROOT/bins/$CONFIG:$ROOT/bins/$CONFIG/protobuf:$PATH
-export CFLAGS="-I$ROOT/include -std=gnu99"
-export LDFLAGS="-L$ROOT/libs/$CONFIG"
+export CFLAGS="-I$ROOT/include -std=gnu99 -fno-wrapv"
 export GRPC_PYTHON_BUILD_WITH_CYTHON=1
-export GRPC_PYTHON_USE_PRECOMPILED_BINARIES=0
 
-if [ "$CONFIG" = "gcov" ]
-then
-  export GRPC_PYTHON_ENABLE_CYTHON_TRACING=1
+# If ccache is available, use it... unless we're on Mac, then all hell breaks
+# loose because Python does hacky things to support other hacky things done to
+# hacky things on Mac OS X
+PLATFORM=`uname -s`
+if [ "${PLATFORM/Darwin}" = "$PLATFORM" ]; then
+  # We're not on Darwin (Mac OS X)
+  if [ -x "$(command -v ccache)" ]; then
+    if [ -x "$(command -v gcc)" ]; then
+      export CC='ccache gcc'
+    elif [ -x "$(command -v clang)" ]; then
+      export CC='ccache clang'
+    fi
+  fi
 fi
 
-tox -e ${TOX_PYTHON_ENV} --notest
+# Find `realpath`
+if [ -x "$(command -v realpath)" ]; then
+  export REALPATH=realpath
+elif [ -x "$(command -v grealpath)" ]; then
+  export REALPATH=grealpath
+else
+  echo 'Couldn'"'"'t find `realpath` or `grealpath`'
+  exit 1
+fi
 
-# We force the .so naming convention in PEP 3149 for side by side installation support
-# Note this is the default in Python3, but explicitly disabled for Darwin, so we only
-# use this hack for our testing environment.
-if [ "$PY_VERSION" -gt "27" ]
-then
-  mv $ROOT/src/python/grpcio/grpc/_cython/cygrpc.so $ROOT/src/python/grpcio/grpc/_cython/cygrpc.so.backup || true
+if [ "$CONFIG" = "gcov" ]; then
```

Comment:
Measuring coverage is broken currently, anyway. I'd be a fan of just removing gcov for now.